### PR TITLE
Added new flags for http timeouts

### PR DESCRIPTION
**Pull request by jcortejoso**
### Description

Added flags to configure http timeouts.

### CODE REVIEW

**Comment by piersy** [comment]
Coverage from tests in `./e2e_test/...` for `./consensus/istanbul/...` at commit 0e46eade95c2fba3da900313120d2fb9bae4926d
<details><summary>coverage:  45.3% of statements across all listed packages</summary><br>
<pre>
coverage:  54.6% of statements in consensus/istanbul
coverage:  45.2% of statements in consensus/istanbul/backend
coverage:   0.0% of statements in consensus/istanbul/backend/backendtest
coverage:  43.3% of statements in consensus/istanbul/backend/internal/db
coverage:  24.1% of statements in consensus/istanbul/backend/internal/enodes
coverage:  22.4% of statements in consensus/istanbul/backend/internal/replica
coverage:  66.0% of statements in consensus/istanbul/core
coverage:   0.0% of statements in consensus/istanbul/proxy
coverage:  75.3% of statements in consensus/istanbul/uptime
coverage: 100.0% of statements in consensus/istanbul/uptime/store
coverage:  51.8% of statements in consensus/istanbul/validator
coverage:  79.2% of statements in consensus/istanbul/validator/random
</pre>
CommentID: 70dace97ff
</details>

File: cmd/utils/flags.go [hunk 1]
```diff
@@ -507,6 +507,21 @@ var (
 		Usage: "HTTP path path prefix on which JSON-RPC is served. Use '/' to serve on all paths.",
 		Value: "",
 	}
+	HTTPRequestReadTimeout = cli.IntFlag{
+		Name:  "http.timeout.read",
+		Usage: "Timeout for HTTP-RPC read requests",
```
Comment (piersy) [inline cmd/utils/flags.go:512]:
```suggestion
		Usage: "Timeout in seconds for HTTP-RPC read requests",
```

File: cmd/utils/flags.go [hunk 2]
```diff
@@ -507,6 +507,21 @@ var (
...
+		Value: node.DefaultHTTPTimeoutRead,
+	}
+	HTTPRequestWriteTimeout = cli.IntFlag{
+		Name:  "http.timeout.write",
+		Usage: "Timeout for HTTP-RPC write requests",
```
Comment (piersy) [inline cmd/utils/flags.go:517]:
```suggestion
		Usage: "Timeout in seconds for HTTP-RPC write requests",
```

File: cmd/utils/flags.go [hunk 3]
```diff
@@ -507,6 +507,21 @@ var (
...
+		Value: node.DefaultHTTPTimeoutWrite,
+	}
+	HTTPRequestIdleTimeout = cli.IntFlag{
+		Name:  "http.timeout.idle",
+		Usage: "Timeout for HTTP-RPC idle connections",
```
Comment (piersy) [inline cmd/utils/flags.go:522]:
```suggestion
		Usage: "Timeout in seconds for HTTP-RPC idle connections",
```

**Review by piersy — APPROVED** [review]
Hi @jcortejoso, i left three requested changes for the flag descriptions, otherwise looks good!

File: cmd/utils/flags.go [hunk 4]
```diff
@@ -507,6 +507,21 @@ var (
 		Usage: "HTTP path path prefix on which JSON-RPC is served. Use '/' to serve on all paths.",
 		Value: "",
 	}
+	HTTPRequestReadTimeout = cli.IntFlag{
+		Name:  "http.timeout.read",
+		Usage: "Timeout for HTTP-RPC read requests",
+		Value: node.DefaultHTTPTimeoutRead,
```
Comment (gastonponti) [inline cmd/utils/flags.go:513]: Instead of using this and duplicate the value for that new node variable, why don't you use instead:
`rpc.DefaultHTTPTimeouts.ReadTimeout`
and include the rpc import?
That ReadTimeout is already multiplied by the seconds, but we can divide it here by `time.Second`
This way, if we change the defaults (for some reason) in the future, everything will be correct

File: node/defaults.go [hunk 5]
```diff
@@ -28,12 +28,15 @@ import (
...
-	DefaultWSPort      = 8546        // Default TCP port for the websocket RPC server
-	DefaultGraphQLHost = "localhost" // Default host interface for the GraphQL server
-	DefaultGraphQLPort = 8547        // Default TCP port for the GraphQL server
+	DefaultHTTPHost         = "localhost" // Default host interface for the HTTP RPC server
+	DefaultHTTPPort         = 8545        // Default TCP port for the HTTP RPC server
+	DefaultWSHost           = "localhost" // Default host interface for the websocket RPC server
+	DefaultWSPort           = 8546        // Default TCP port for the websocket RPC server
+	DefaultGraphQLHost      = "localhost" // Default host interface for the GraphQL server
+	DefaultGraphQLPort      = 8547        // Default TCP port for the GraphQL server
+	DefaultHTTPTimeoutRead  = 30          // Default HTTP Read request timeout in seconds
```
Comment (gastonponti) [inline node/defaults.go:37]: Another possibility is to change it here, but that will generate a variable that references another variable which makes de code a little harder to read.
But as is it, if we change the defaults, we will have to remember to also change these values, that's why I prefer the other way

File: cmd/utils/flags.go [hunk 6]
```diff
@@ -507,6 +508,21 @@ var (
 		Usage: "HTTP path path prefix on which JSON-RPC is served. Use '/' to serve on all paths.",
 		Value: "",
 	}
+	HTTPRequestReadTimeout = cli.IntFlag{
+		Name:  "http.timeout.read",
+		Usage: "Timeout in seconds for HTTP-RPC read requests",
+		Value: int(rpc.DefaultHTTPTimeouts.ReadTimeout),
```
Comment (gastonponti) [inline cmd/utils/flags.go:514]: If the message timeout is in seconds, this `value` should be `int(rpc.DefaultHTTPTimeouts.ReadTimeout / Time.Second)`

File: cmd/utils/flags.go [hunk 7]
```diff
@@ -507,6 +508,21 @@ var (
...
+	}
+	HTTPRequestWriteTimeout = cli.IntFlag{
+		Name:  "http.timeout.write",
+		Usage: "Timeout in seconds for HTTP-RPC write requests",
+		Value: int(rpc.DefaultHTTPTimeouts.WriteTimeout),
```
Comment (gastonponti) [inline cmd/utils/flags.go:519]: same

File: cmd/utils/flags.go [hunk 8]
```diff
@@ -507,6 +508,21 @@ var (
...
+	}
+	HTTPRequestIdleTimeout = cli.IntFlag{
+		Name:  "http.timeout.idle",
+		Usage: "Timeout in seconds for HTTP-RPC idle connections",
+		Value: int(rpc.DefaultHTTPTimeouts.IdleTimeout),
```
Comment (gastonponti) [inline cmd/utils/flags.go:524]: same

File: eth/protocols/snap/sync.go [hunk 9]
```diff
@@ -509,7 +509,7 @@ func (s *Syncer) Unregister(id string) error {
 	// Remove all traces of the peer from the registry
 	s.lock.Lock()
 	if _, ok := s.peers[id]; !ok {
-		log.Error("Snap peer not registered", "id", id)
+		log.Error("Snap ", "id", id)
```
Comment (gastonponti) [inline eth/protocols/snap/sync.go:512]: probably not part of this PR

Comment (gastonponti) [inline eth/protocols/snap/sync.go:512]:
```suggestion
		log.Error("Snap peer not registered", "id", id)
```
